### PR TITLE
[WIP] Shared completion for Group Dailies

### PR DESCRIPTION
**Pull request by paglias**
TODO 

- [ ] tests

-------------

This is the continuation of https://github.com/HabitRPG/habitica/pull/11179 and now depends on #11389

### CODE REVIEW

**Comment by paglias** [comment]
Hmm I think there've been a few problems with merging here? because there seem to be a lot of unrelated changes

File: website/client/components/tasks/task.vue [hunk 1]
```diff
@@ -11,7 +11,7 @@
       // Dailies and todos left side control
       .left-control.d-flex.justify-content-center(v-if="task.type === 'daily' || task.type === 'todo'", :class="[{'control-bottom-box': this.task.group.id, 'control-top-box': approvalsClass}, controlClass.bg]")
         .task-control.daily-todo-control(:class="controlClass.inner", @click="isUser ? score(task.completed ? 'down' : 'up') : null")
-          .svg-icon.lock(v-html="icons.lock", v-if="this.task.group.id && !isUser && !task.completed", :class="controlClass.icon")
+          .svg-icon.lock(v-html="icons.lock", v-if="this.task.group.id && !isUser && !task.completed", :class="controlClass.checkbox")
```
Comment (paglias) [inline website/client/components/tasks/task.vue]: I don't think that controlClass.checkbox exist as only the value but not the key was changed so it's always controlClass.icon

**Comment by paglias** [comment]
The conflicts should be fixed but as there were many I might have messed up something

**Comment by MeanderingCode** [comment]
I'm sorry I disappeared on this. My life became overloaded.

If there are any issues, I will try to dig in. I'm going to assume that it's handled for now.
Once it's deployed, I will revitalize my family's use of Habitica and will put it thoroughly to the test :)
Also, my life should be stabilizing a bit soon, and that may give me time to add some tests and add some of the enhancements I wanted, and fix bugs.

Thanks for keeping it up!! I'm excited to use it.

**Comment by MeanderingCode** [comment]
Hello again, all!

I'm not sure where things are with this based on the tag changes, but I'd like to help push this over the finish line. Can Someone give me some insight into what's next?

**Comment by SabreCat** [comment]
Oh, welcome back @MeanderingCode! We put this on the testing servers a while back, but it didn't appear to be working--the tasks stayed uncompleted no matter their configuration or the workflow you followed.

**Comment by MeanderingCode** [comment]
Interesting. I wonder if there was a change from `develop` that effected this.
I'll pull down the branch this week and poke around.

**Comment by MeanderingCode** [comment]
I have pulled down this branch with others' changes, tested it, merged in the latest develop and resolved conflicts, and tested that.

It appears to still work for me. Can someone chime in about what didn't work?

(Also, I pushed my merge of develop and another eslint commit and I do not see this PR updated)

**Comment by MeanderingCode** [comment]
To be clear, my local testing yielded the following in a Group with two users, all tasks assigned to both users:

* Single completion Group Daily:
  * UserA completes the Daily: Group and UserB show Daily as complete
* All completion Group Daily:
  * UserA completes the Daily: No change in Group or UserB
  * UserB completes the Daily: Group and UserA show Daily complete
* Single completion Todo:
  * UserA completes Todo: Group shows Todo complete, UserB Todo deleted
* All completion Todo:
  * UserA completes Todo: No change in Group or UserB
  * UserB completes Todo: Group and Users show Todo complete

**Comment by MeanderingCode** [comment]
I would really like to see this completed and released. Is anyone willing to reexamine this?

**Comment by MeanderingCode** [comment]
It also occurs to me that there is a possibility that there is a difference in behavior between development and production builds.

I will attempt to rule this out or validate it by producing a production build, but perhaps someone more savvy with javascript could assist with this, since I will be attempting this from context I can find in `package.json` and the Docker and compose files.

**Comment by paglias** [comment]
@MeanderingCode we're willing to re-examine the PR and will do but it might take a few days. Regarding differences between production and development they should not affect this code, just make sure to rebuild your docker image with the new code

**Comment by SabreCat** [comment]
OK, I ran through this today and was able to complete the workflows! I wonder if my previous looks had bad merge conflict resolution, or the like, that prevented it from functioning as intended. I'm going to put this back in the testing environment. @paglias do you have any objections re the code itself?

**Comment by SabreCat** [comment]
I've made an attempt at removing unrelated changes in a commit above, but in the process I got a much better idea of where exactly things went pear-shaped, so I'm planning to go back through tomorrow with a slightly different strategy that might be safer.

**Comment by SabreCat** [comment]
FYI @paglias: It looks like at some point, you merged @phillipthelen's bulk task scoring PR into here, possibly in an effort to resolve merge conflicts on Delta, which led to the current mess. I'm going to try rewinding to @MeanderingCode's last intentional change at https://github.com/HabitRPG/habitica/pull/11390/commits/bf87d6689a8c7cb943de653b89e6fb16f1d9f654, then apply our subsequent fixes and test additions without the rogue merge, probably in a separate PR.

**Comment by MeanderingCode** [comment]
I'm glad it works!

@SabreCat Please note that there are three further "intentional" commits since https://github.com/HabitRPG/habitica/commit/bf87d6689a8c7cb943de653b89e6fb16f1d9f654 .

They are:
* https://github.com/HabitRPG/habitica/commit/30ac2f95193d3af918d2e4ba7f4adb2f470918b4
  `eslint became strict`
* https://github.com/HabitRPG/habitica/commit/536d34e2a9c7ddfba6a3685d21c8a2ca154abed9
  `Improve group daily cron uncompletion from suggestions`
* https://github.com/HabitRPG/habitica/commit/a8608758870d6fe40c0bb1b93b2167f84a3e2e5b
  `Catch edge case where user uncompletes a group shared task they never completed`

I'm not sure why pushing them to my branch did not update this PR, but I suspect it has to do with GH's behavior around copying branches across forks and edit permissions on PRs for non-openers and non-maintainers.

**Comment by paglias** [comment]
@SabreCat yeah I think I had to merge this on top the bulk scoring PR to fix some conflicts but it can be removed, the bulk scoring is currently on hold so we can go ahead with this one and revisit the other later

**Comment by SabreCat** [comment]
Continued in #12219, phew!

**Pull request closed without merging**